pallets/phala/src/stakepool.rs

```diff
@@ -791,6 +767,30 @@ pub mod pallet {
 		withdraw_queue: VecDeque<WithdrawInfo<AccountId, Balance>>,
 	}
 
+	impl<AccountId, Balance> PoolInfo<AccountId, Balance>
+	where
+		AccountId: Default,
+		Balance: sp_runtime::traits::AtLeast32BitUnsigned + Copy,
+	{
+		/// Clears the pending rewards of a user and move to `available_rewards` for claiming
+		fn clear_user_pending_reward(&self, user_info: &mut UserStakeInfo<AccountId, Balance>) {
+			let pending_reward = user_info.pending_reward(self.pool_acc);
+			user_info
+				.available_rewards
+				.saturating_accrue(pending_reward);
+			user_info.clear_pending_reward(self.pool_acc);
```

Comment:
According to the algorithm, looks like we should only reset user debt after amount updated